src/UnitfulRecipes.jl

```diff
@@ -76,25 +77,70 @@ function _resolve_axis!(attr, arr, T, axis)
     end
 
     # get axis label and append unit
-    key = key_label(axis)
-    ustr = string(u)
-    add_unit_if_missing!(attr, key_label(axis), ustr)
+    append_unit_if_needed!(attr, key_guide(axis), u)
 
     # colorbar_title
-    ckey = (axis==3) ? :colorbar_title : key_label(axis)
-    add_unit_if_missing!(attr, ckey, ustr)
+    if axis == 3
+        append_unit_if_needed!(attr, :colorbar_title, u)
+    end
 
     return arr
 end
 
-function add_unit_if_missing!(attr, key, ustr::AbstractString)
-    value = get!(attr, key, ustr)
-    if !occursin(ustr, value)
-        attr[key] = "$value $(ustr)"
+abstract type AbstractProtectedString <: AbstractString end
+"""
+    ProtectedString
+
+Wrapper around a `String` to "protect" it from `recipe!` passes.
+"""
+struct ProtectedString <: AbstractProtectedString
+    content::String
+end
+struct UnitfulString{U} <: AbstractProtectedString
+    content::String
+    unit::U
+end
+# Minimum required AbstractString interface to work with Plots?
+const S = AbstractProtectedString
+Base.iterate(n::S) = iterate(n.content)
+Base.iterate(n::S, i::Integer) = iterate(n.content, i)
+Base.codeunit(n::S) = codeunit(n.content)
+Base.ncodeunits(n::S) = ncodeunits(n.content)
+Base.isvalid(n::S, i::Integer) = isvalid(n.content, i)
+Base.pointer(n::S) = pointer(n.content)
+Base.pointer(n::S, i::Integer) = pointer(n.content, i)
+# macro for easy-to-use interface?
+# i.e., so that `P"foo"` creates `ProtectedString("foo")`
+macro P_str(s)
+    return ProtectedString(s)
+end
+
+
+function append_unit_if_needed!(attr, key, u::Unitful.Units)
+    label = get(attr, key, nothing)
+    append_unit_if_needed!(attr, key, label, u)
+    return attr
+end
+# dispatch on the type of `label`
+append_unit_if_needed!(attr, key, label::ProtectedString, u) = attr
+function append_unit_if_needed!(attr, key, label::Nothing, u)
+    attr[key] = UnitfulString(string(u), u)
+    return attr
+end
+function append_unit_if_needed!(attr, key, label::UnitfulString, u)
+    (label.unit ≠ u) && error("The unit of $key has changed!")
+    return attr
+end
+function append_unit_if_needed!(attr, key, label::String, u)
+    if label ≠ ""
+        attr[key] = UnitfulString(string(label, " (", u, ")"), u)
```

Comment:
In my experience `[kg]` is more common then `(kg)`, but I can live with both variants.

Reply:
Oh yes sorry, I don't care either, it's just that I started by copying what I had before. Maybe we can have an enironment variable to set for either of the two?

Reply:
I think we should stick to one instead of introducing this extra complexity. I am okay with `(kg)`.

Reply:
OK no worries, I'll leave it as is and we can always change it later!